src/utils/style.ts

```diff
@@ -25,16 +26,39 @@ export const absoluteCenter = css`
   transform: translate(-50%, -50%);
 `;
 
-export const isValidCSSHex = (hex: string) => {
+export const isValidCSSHex = (hex: string): boolean => {
   // matches 6 digit characters prefixed with a '#'.
   return /^#[0-9A-F]{6}$/i.test(hex);
 };
 
-export const hexTorgba = (hex: string, alpha: number = 1) => {
+export const hexTorgba = (hex: string, alpha: number = 1): string => {
   if (!isValidCSSHex(hex)) {
     return '';
   }
 
   const [r, g, b]: any = hex.match(/\w\w/g)?.map((h) => parseInt(h, 16));
   return `rgba(${r}, ${g}, ${b}, ${alpha || 1})`;
 };
+
+// There's a breaking change in styled-components v6 - shouldForwardProp is no longer provided by default
+// https://styled-components.com/docs/faqs#shouldforwardprop-is-no-longer-provided-by-default
+// This maintains backwards compatibility with v5 by filtering props using @emotion/is-prop-valid
+// https://styled-components.com/docs/faqs#shouldforwardprop-is-no-longer-provided-by-default
+
+/**
+ * Creates a configuration object for styled-components to control prop forwarding
+ * @param additionalProps - Array of custom prop names that should be forwarded
+ * @returns Configuration object with shouldForwardProp function
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const createStyledConfig = (additionalProps: string[] = []) => ({
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  shouldForwardProp: (prop: string) => {
+    // Forward custom props
+    if (additionalProps.includes(prop)) return true;
+
+    return isPropValid(prop);
```

Comment:
For nested styled components, we add withConfig only to the bottom most one for the filtering. This way it won't block any react props being passing through a styled components.

For example:
```
const StyledVideoTile = styled.div.withConfig()`...`;

// React component 
const VideoTile = (props) => {
  return props.isActive? <StyledVideoTile/> : null
}

// Styled react component
const ContentTile = styled(VideoTile)`...`; // Do not use withConfig, because StyledVideoTile already have it, this way isActive will not be filtered.

<StyledContentTile isActive={true} />
```